src/Components/AllCandidates/AllCandidates.jsx

```diff
@@ -135,66 +171,121 @@ export default class AllCandidates extends Component {
     this.props.history.push(path);
   }
 
+  // Get comments from users on each application
+  getComments = () => {
```

Comment:
Is this function just extracting all the comments of an application for a CSV? Looks like youre trying to get an array of application comments ordered by application, but correct me if im wrong.
If that's the case then can clean up this function, see branch xin/refactor-all-candidates for context, I mostly am just talking about avoiding so much nested looping.

One Suggestion:
Create map of application id : comments
For each review, if the application id of the review is a key in the map, then we extract the comments of the review and append them to the map entry
Concat all map entries to get one array

Also can we move the associating user's email with review to backend? 
I think it makes sense in the future to just store the user's email along with the review as well.

Since I have a branch to turn this into a func component that also refactors the avg ranking calc already, feel free to delegate this refactor to me as well

Reply:
Yeah it does get comments for each individual application, like this export:
![image](https://user-images.githubusercontent.com/13268990/89109871-129d7e80-d413-11ea-8937-5b0c7b9dffdf.png)

For now, I think we want this export logic included in the next push to master, and I don't think I can get this done and tested in time. If this is something you can take on along with refactoring the avg ranking calculations, that would be great. Otherwise I'll do it later.